### PR TITLE
Remove backoff and deadletter types

### PR DESCRIPTION
**Pull request by povilasv**
This simplifies creating new back off retrying and dead letter handlers.
```
source.ConsumeMessages(ctx, handler, pubsub.ConsumerErrorHandler(backoff.New(handler, 3)))
```
becomes:
```
source.ConsumeMessages(ctx, handler, backoff.New(handler, 3))
```

### CODE REVIEW

**Comment by jakekeeys** [comment]
LGTM but this will break current usages

File: consumer/deadletter/deadletter.go [hunk 1]
```diff
@@ -30,7 +29,7 @@ func New(sink pubsub.MessageSink, messageTopic string) JSONDeadLetteringErrorHan
 }
 
 // NewWithFallback returns a new JSONDeadLetteringErrorHandler with fallback
```
Comment (thinktainer) [inline consumer/deadletter/deadletter.go]: JSON seems to no longer apply

**Comment by povilasv** [comment]
@jakekeeys 

This shouldn't break changes as it is safe to do `pubsub.ConsumerErrorHandler()` on a `ConsumerErrorHandler` type. And my guess is that most users who use `backoff.New()` use it with `ConsumeMessages()` method of pubsub

File: consumer/deadletter/deadletter.go (hunk 1 above)
Comment (povilasv) [reply consumer/deadletter/deadletter.go]: Thanks for noting, fixed.

**Pull request merged**